src/api-specification/contribution-controller/retrieving-a-contribution-by-ID.md

```diff
@@ -0,0 +1,102 @@
+# Retrieving a contribution by ID
+
+> Retrieves a contribution that is stored on the network by its [`id`](../../overview/contributions.md#retrieving-contributions) (Fraud Identifier).\
+> The `assetDefinitionId` value of any given retrieved contribution may be used to [assemble a contribution flag](assembling-a-contribution-flag.md) for it.
+
+**Protocol**: `HTTP`
+
+**Encoding**: `JSON`
+
+**Endpoint**: `/data/api/v1/contribution-management/contribution/{id}`
+
+**Method**: `GET`
+
+**Header**: `Authorization` — represented by a `string` that is the `accessToken` value that was returned either at the authorization, or the previous time the token [has been refreshed](refreshing-authentication-tokens.md).
+
+**Parameters**:
+
+| Field | Value Type     | Description                                                                                                            |
+| :---: | -------------- | ---------------------------------------------------------------------------------------------------------------------- |
+| `id`  | `string(path)` | Range of [`id` values](../../overview/contributions.md#retrieving-contributions) (Fraud Identifiers), or a single one. |
+
+#### Input/request structure
+
+A `GET` request to the endpoint with the `{id}` parameter specified.
+
+**Example**: `data/api/v1/subscription-management/subscription/192.168.1.1`
+
+#### Output/response structure
+
+::: details Show
+
+::: code-group Data structure
+
+```json5 [Structure]
+{
+  "status": {
+    "code": integer($int32),
+    "name": "string",
+    "message": "string"
+  },
+  "data": [
+    {
+        "assetDefinitionIds": "idRange#contribution",
+        "contribution": {
+            "id": "idRange", //127.0.0.1 OR 127.0.0.1-127.0.0.2 OR +14155552671 OR +14155552671-+14155552672 OR 107615702016566
```

Comment:
from here and potentially other places, applicable to any attribute, not just id